pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -193,10 +195,24 @@ func buildLogsTimeSeriesFilterQuery(fs *v3.FilterSet, groupBy []v3.AttributeKey,
 				case v3.FilterOperatorContains, v3.FilterOperatorNotContains:
 					columnName := getClickhouseColumnName(item.Key)
 					val := utils.QuoteEscapedString(fmt.Sprintf("%v", item.Value))
-					conditions = append(conditions, fmt.Sprintf("%s %s '%%%s%%'", columnName, logsOp, val))
+					if columnName == BODY {
+						logsOp = strings.Replace(logsOp, "ILIKE", "LIKE", 1) // removing i from ilike and not ilike
+						conditions = append(conditions, fmt.Sprintf("lower(%s) %s lower('%%%s%%')", columnName, logsOp, val))
```

Comment:
I would use some function on the `logsOp` itself like `logsOp.<func>()`  rather than the `strings.Replace`.

Reply:
but the implementation will remain the same right ? or you mean somethig else ?

Reply:
We support case-insensitive operations but this change doesn't account for that. If you want to account for that you can either add a statement here and below or do all of the logic separately that explicitly highlights why we do it and use that function here.